app/javascript/src/components/Navigation/components/NotificationImage.jsx

```diff
@@ -1,8 +1,47 @@
 import React from "react";
+import { Image } from "react-bootstrap";
+import PropTypes from "prop-types";
 
-export default () =>
-  <div className="text-center">
-    <span className="assignment">
-      <i className="fa fa-newspaper-o" />
-    </span>
-  </div>;
+const NotificationImage = ({className, type, avatar}) => {
+  const delegator = {
+      recent_changes: (
+          <Image
+              className="avatar"
+              src={`${avatar }?${new Date().getTime()}`}
+              circle
+          />
+      ),
+      system_message: (
+          <span className="system-message">
+              <i className="glyphicon glyphicon-tower" />
+          </span>
+      ),
+      deliver: (
+          <span className="deliver">
+              <i className="fa fa-truck" />
+          </span>
+      ),
+      assignment: (
+          <span className="assignment">
+              <i className="fa fa-newspaper-o" />
+          </span>
+      )
+  }
+  return (
+    <div className={`text-center ${className}`} >
+      {delegator[type]}
+    </div>
+  );
+};
+
+NotificationImage.defaultProps = {
```

Comment:
So we use flow now for type checking is a banana I know but we decided to use it because now is fancy to have static type checking in JS nowadays. Check this link on how to use this https://flow.org/
I also installed flo-bro (https://www.npmjs.com/package/flow-bro) it runs a watcher that checks if your types are valid

Reply:
So no more propTypes?